nutils/evaluable.py

```diff
@@ -4164,14 +4168,18 @@ def _serialized(self):
         indices = {d: i for i, d in enumerate(itertools.chain([self._index], self._invariants, self._dependencies))}
         return tuple((dep, tuple(map(indices.__getitem__, dep._Evaluable__args))) for dep in self._dependencies)
 
+    @property
+    def _serialized_evalf(self):
+        return tuple((dep.evalf, indices) for dep, indices in self._serialized)
```

Comment:
We are propagating the situation here of two identical `_serialized_evalf` implementations mirrorring near-identical implementations of `_serialized`. Could we consider an index-holding base class here for LoopSum and LoopConcatenateCombined?

Reply:
I'd like the idea of merging common parts of `LoopSum` and `LoopConcatenateCombined`, but propose to defer this to a separate PR. I'd like to finish #700 first.